src/node/channels.h

```diff
@@ -917,14 +928,32 @@ namespace ccf
           message_limit);
         channels.emplace_hint(search, peer_id, std::move(channel));
       }
-      else if (!search->second->is_outgoing())
+      else if (search->second && !search->second->is_outgoing())
       {
         // Channel with peer already exists but is incoming. Create host
         // outgoing connection.
         LOG_DEBUG_FMT("Setting existing channel to {} as outgoing", peer_id);
         search->second->set_outgoing(hostname, service);
         return;
       }
+      else if (!search->second)
+      {
+        LOG_DEBUG_FMT(
+          "Re-creating new outbound channel to {} ({}:{})",
+          peer_id,
+          hostname,
+          service);
+        search->second = std::make_shared<Channel>(
```

Comment:
@jumaffre I'm not 100% sure we need this branch, but if channels can be closed and re-opened later, then we do.